enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingService.java

```diff
@@ -316,6 +318,58 @@ public void terminateInactiveProcesses() {
         otpService.terminateOldOtps(createdDateOtp);
     }
 
+    /**
+     * Provide consent text.
+     *
+     * @param request consent text request
+     * @return consent response
+     */
+    public OnboardingConsentTextResponse fetchConsentText(final OnboardingConsentTextRequest request) throws OnboardingProcessException {
+        final ConsentTextRequest providerRequest = ConsentTextRequest.builder()
+                .processId(request.getProcessId())
+                .userId(request.getUserId())
+                .consentType(convert(request.getConsentType()))
+                .locale(LocaleUtils.toLocale(request.getLanguage()))
+                .build();
+
+        try {
+            final String consentText = onboardingProvider.fetchConsent(providerRequest);
+            final OnboardingConsentTextResponse response = new OnboardingConsentTextResponse();
+            response.setConsentText(consentText);
+            return response;
+        } catch (OnboardingProviderException e) {
+            throw new OnboardingProcessException("An error when fetching consent text.", e);
+        }
+    }
+
+    /**
+     * Record dis/approval of consent
+     *
+     * @param request approval request
+     */
+    public void approveConsent(final OnboardingConsentApprovalRequest request) throws OnboardingProcessException {
+        final ApproveConsentRequest providerRequest = ApproveConsentRequest.builder()
+                .processId(request.getProcessId())
+                .userId(request.getUserId())
+                .consentType(convert(request.getConsentType()))
+                .approved(request.getApproved())
+                .build();
+
+        try {
+            onboardingProvider.approveConsent(providerRequest);
+        } catch (OnboardingProviderException e) {
+            throw new OnboardingProcessException("An error when approving consent.", e);
+        }
+    }
+
+    private static com.wultra.app.onboardingserver.provider.ConsentType convert(final ConsentType source) {
```

Comment:
This seems like duplicity in `ConsentType` but I want to isolate possible changes in REST a provider API.

Reply:
this method should be isolated to separate converter class.